tests/functional/component/test_component.py

```diff
@@ -0,0 +1,130 @@
+#
+# -----------------------------------------------------------------------------
+#                     The CodeChecker Infrastructure
+#   This file is distributed under the University of Illinois Open Source
+#   License. See LICENSE.TXT for details.
+# -----------------------------------------------------------------------------
+
+import os
+import unittest
+
+from shared.ttypes import Permission
+
+from codeCheckerDBAccess_v6.ttypes import ReportFilter
+
+from libtest import env
+
+
+class TestComponent(unittest.TestCase):
+
+    def setUp(self):
+        self._test_workspace = os.environ.get('TEST_WORKSPACE')
+
+        test_class = self.__class__.__name__
+        print('Running ' + test_class + ' tests in ' + self._test_workspace)
+
+        self._clang_to_test = env.clang_to_test()
+
+        self._testproject_data = env.setup_test_proj_cfg(self._test_workspace)
+        self.assertIsNotNone(self._testproject_data)
+
+        # Get the test configuration from the prepared int the test workspace.
+        self.test_cfg = env.import_test_cfg(self._test_workspace)
+
+        self._product_name = self.test_cfg['codechecker_cfg']['viewer_product']
+        pr_client = env.setup_product_client(
+            self._test_workspace, product=self._product_name)
+        product_id = pr_client.getCurrentProduct().id
+
+        # Setup an authentication client for creating sessions.
+        self._auth_client = env.setup_auth_client(self._test_workspace,
+                                                  session_token='_PROHIBIT')
+
+        # Create an PRODUCT_ADMIN login.
+        admin_token = self._auth_client.performLogin("Username:Password",
+                                                     "admin:admin123")
+
+        extra_params = '{"productID":' + str(product_id) + '}'
+        ret = self._auth_client.addPermission(Permission.PRODUCT_ADMIN,
+                                              "admin",
+                                              False,
+                                              extra_params)
+        self.assertTrue(ret)
+
+        self._cc_client = env.setup_viewer_client(self._test_workspace,
+                                                  session_token=admin_token)
+        self.assertIsNotNone(self._cc_client)
+
+        # Get the run names which belong to this test
+        run_names = env.get_run_names(self._test_workspace)
+
+        runs = self._cc_client.getRunData(None)
+
+        test_runs = [run for run in runs if run.name in run_names]
+
+        self.assertEqual(len(test_runs), 1,
+                         'There should be only one run for this test.')
+        self._runid = test_runs[0].runId
+        self._run_name = test_runs[0].name
+
+        self._component_name = 'dummy_component'
```

Comment:
Do we handle component names with spaces and utf8 characters right? Please add tests for them.

Reply:
- I created a separate test case for white spaces.
- I handled utf-8 errors on the client side.